src/Expander/DeclScope.hs

```diff
@@ -2,6 +2,13 @@ module Expander.DeclScope where
 
 import Data.Unique
 
+-- | A 'DeclValidityPtr' gets filled with a 'ScopeSet' representing an
+-- environment. Each 'Scope' in this 'ScopeSet' corresponds to an identifier
```

Comment:
I don't think `ScopeSets` represent environments *per se*. They instead represent scopes in the abstract sense, rather than in the sense of the specific datatype we have here - remember, they are generalizations of de Bruijn indices that account for multiple dimensions of binding induced by macro expansion. The difference is that environments are fundamentally mappings from names to values, while these are just about which names can be referred to by a given identifier.

Reply:
I have now changed the comment to avoid misusing the word "environment".